src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,372 @@
+/* SPDX-License-Identifier: LGPL-2.1+
+ * Copyright © 2020 VMware, Inc. */
+
+#include <linux/mptcp.h>
+
+#include "alloc-util.h"
+#include "netlink-util.h"
+#include "networkd-manager.h"
+#include "networkd-mptcp.h"
+#include "parse-util.h"
+#include "set.h"
+#include "string-util.h"
+
+static int mp_tcp_new(MPTCP **ret) {
+        MPTCP *mp_tcp;
+
+        mp_tcp = new(MPTCP, 1);
+        if (!mp_tcp)
+                return -ENOMEM;
+
+        *mp_tcp = (MPTCP) {
+                  .subflows = (uint32_t) -1,
+                  .add_addr_accepted = (uint32_t) -1,
+                  .family = AF_UNSPEC,
+                  .id = (uint8_t) -1,
```

Comment:
Indentation?

Why `id` is set to `-1`? `255` seems a valid value for `id`. I think it is better to introduce `id_is_set` boolean element if it is mandatory to specify.